google/cloud/internal/opentelemetry.h

```diff
@@ -84,9 +87,57 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
         attributes);
 
 /**
- * Extracts information from a `Status` and adds it to a span.
+ * Start a span with a @p name, @p attributes using an initializer list, and @p
+ * links using an initializer lists.
+ */
+opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
```

Comment:
These new constructors look unnecessarily complicated to me. Do we use them all?

The less interfaces the better, even in our internal code.

Reply:
I agree. This is what otel does to get the pattern matching. We do not use all the constructors in other places, but some of these constructors call each other. 

I also do not think this is ideal.